### PR TITLE
[MLIR][TORCH] Add support for integer type input for `max.dim` op

### PR DESCRIPTION
**Pull request by vivekkhandelwal1**
Signed-Off By: Vivek Khandelwal <vivek@nod-labs.com>

### CODE REVIEW

**Comment by vivekkhandelwal1** [comment]
Hi @ramiro050, can you review this patch as well?

File: python/torch_mlir_e2e_test/test_suite/reduction.py [hunk 1]
```diff
@@ -315,6 +315,44 @@ def ReduceMaxAlongDim_basic(module, tu: TestUtils):
...
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1, -1], torch.int64, True),
+    ])
+    def forward(self, a):
+        return torch.ops.aten.max(a, 1)[0]
```
Comment (ramiro050) [inline python/torch_mlir_e2e_test/test_suite/reduction.py:328]: Why only return the first element of the result?

Comment (vivekkhandelwal1) [reply python/torch_mlir_e2e_test/test_suite/reduction.py]: Fixed

File: python/torch_mlir_e2e_test/test_suite/reduction.py [hunk 2]
```diff
@@ -315,6 +315,44 @@ def ReduceMaxAlongDim_basic(module, tu: TestUtils):
...
+        None,
+        ([-1, -1, -1], torch.uint8, True),
+    ])
+    def forward(self, a):
+        return torch.ops.aten.max(a, 1)
+
+
+@register_test_case(module_factory=lambda: ReduceMaxAlongDimUnsignedInt())
+def ReduceMaxAlongDimUnsignedInt_basic(module, tu: TestUtils):
+    module.forward(tu.randint(3, 4, 5, high=100).to(torch.uint8))
```
Comment (ramiro050) [inline python/torch_mlir_e2e_test/test_suite/reduction.py:352]: nit: can you make the ranges for both tests include negative numbers?

**Review by ramiro050 — APPROVED** [review]
LGTM, just one small change request

File: python/torch_mlir_e2e_test/test_suite/reduction.py (hunk 2 above)
Comment (vivekkhandelwal1) [reply python/torch_mlir_e2e_test/test_suite/reduction.py]: Thanks for this comment! It resulted in a small fix in the implementation.

**Pull request merged**